File: src/main/java/com/cjie/akka/actor/PingPongApplication.java
package com.cjie.akka.actor;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

/**
 * Created with IntelliJ IDEA.
 * User: hucj
 * Date: 14-10-29
 * Time: 下午12:57
 * To change this template use File | Settings | File Templates.
 */
public class PingPongApplication {

    public static void main(String[] args) throws InterruptedException {
        ActorSystem _system = ActorSystem.create("BecomeUnbecome");
        ActorRef pingPongActor = _system
                .actorOf(new Props(cn.cjie.scala.akka.actor.PingPongActor.class));
        pingPongActor.tell(PingPongActor.PING, pingPongActor);
        Thread.sleep(2000);
        _system.shutdown();
    }
}

File: src/main/scala/cn/cjie/scala/akka/actor/PingPongApplication.scala
package cn.cjie.scala.akka.actor

import akka.actor.{Props, ActorSystem}

/**
 * Created with IntelliJ IDEA.
 * User: hucj
 * Date: 14-10-29
 * Time: 下午1:01
 * To change this template use File | Settings | File Templates.
 */
object PingPongApplication extends App {
  val _system = ActorSystem("BecomeUnbecome")
  val pingPongActor = _system.actorOf(Props[PingPongActor])
  pingPongActor ! PING
  Thread.sleep(2000)
  _system.shutdown
}
